packages/perf/android/src/main/java/io/invertase/firebase/perf/ScreenTrace.java

```diff
@@ -0,0 +1,227 @@
+  public ScreenTrace(Activity activity, String tag) {
+    this.activity = activity;
+    // We don't care about adding the activity name to the trace name
+    // because RN doesn't care about activities
+    this.traceName = tag;
+
+    enableHardwareAcceleration(activity);
+
+    isScreenTraceSupported = isScreenTraceSupported(activity);
+
+    if (isScreenTraceSupported) {
+      frameMetricsAggregator = new FrameMetricsAggregator();
+    }
+  }
+
+  // region Public APIs
+
+  /**
+   * Force enable Hardware acceleration to support screen traces as we can't observe frame
+   * rates for a non hardware accelerated view.
+   * <p>
+   * See: https://developer.android.com/guide/topics/graphics/hardware-accel
+   */
+  public static void enableHardwareAcceleration(Activity activity) {
+    activity.getWindow().setFlags(
+      WindowManager.LayoutParams.FLAG_HARDWARE_ACCELERATED,
+      WindowManager.LayoutParams.FLAG_HARDWARE_ACCELERATED);
+  }
+
+  /**
+   * Returns whether recording of screen traces are supported or not.
+   */
+  public boolean isScreenTraceSupported() {
+    return isScreenTraceSupported;
+  }
+
+  /**
+   * Starts recording the frame metrics for the screen traces.
+   */
+  public void recordScreenTrace() {
+    if (!isScreenTraceSupported) {
+      throw new IllegalArgumentException("Trying to record screen trace when it's not supported!");
+    }
+
+    Log.d(TAG, "Recording screen trace " + traceName);
+
+    frameMetricsAggregator.add(activity);
+    perfScreenTrace = FirebasePerformance.startTrace(getScreenTraceName());
+  }
+
+  /**
+   * Stops recording screen traces and dispatches the trace capturing information on %age of
+   * Slow/Frozen frames.
+   * <p>
+   * Reference: Fireperf Source Code - https://bityl.co/5v22
+   */
+  public void sendScreenTrace() {
+    if (perfScreenTrace == null) return;
+
+    int totalFrames = 0;
+    int slowFrames = 0;
+    int frozenFrames = 0;
+
+    // Stops recording metrics for this Activity and returns the currently-collected metrics
+    SparseIntArray[] arr = frameMetricsAggregator.reset();
+
+    if (arr != null) {
+      SparseIntArray frameTimes = arr[FrameMetricsAggregator.TOTAL_INDEX];
+
+      if (frameTimes != null) {
+        for (int i = 0; i < frameTimes.size(); i++) {
+          int frameTime = frameTimes.keyAt(i);
+          int numFrames = frameTimes.valueAt(i);
+
+          totalFrames += numFrames;
+
+          if (frameTime > Constants.FROZEN_FRAME_TIME) {
+            // Frozen frames mean the app appear frozen. The recommended thresholds is 700ms
+            frozenFrames += numFrames;
+          }
+
+          if (frameTime > Constants.SLOW_FRAME_TIME) {
+            // Slow frames are anything above 16ms (i.e. 60 frames/second)
+            slowFrames += numFrames;
+          }
+        }
+      }
+    }
+
+    if (totalFrames == 0 && slowFrames == 0 && frozenFrames == 0) {
+      // All metrics are zero, no need to send screen trace.
+      // return;
```

Comment:
interesting decision to a) comment out the code versus simply remove it - committing commented out code is usually a bad sign or b) alternatively not document why it is commented out but included

Stated differently: why is this return commented out, contrary to the stated intent of the original comment?

Reply:
fixed

Reply:
this one may still not be resolved, unsure, but there are no major side effects that will impact users and if the metrics are off, we'll have to see folks using it in practice submit fixes I suppose